app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationViewHolder.java

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2017 Andrew Dawson
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.conversation;
+
+import android.content.Context;
+import android.text.InputFilter;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.ToggleButton;
+
+import com.keylesspalace.tusky.R;
+import com.keylesspalace.tusky.adapter.StatusBaseViewHolder;
+import com.keylesspalace.tusky.entity.Attachment;
+import com.keylesspalace.tusky.interfaces.StatusActionListener;
+import com.keylesspalace.tusky.util.SmartLengthInputFilter;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class ConversationViewHolder extends StatusBaseViewHolder {
+    private static final InputFilter[] COLLAPSE_INPUT_FILTER = new InputFilter[]{SmartLengthInputFilter.INSTANCE};
+    private static final InputFilter[] NO_INPUT_FILTER = new InputFilter[0];
+
+    private TextView conversationNameTextView;
+    private ToggleButton contentCollapseButton;
+    private ImageView[] avatars;
+
+    private StatusActionListener listener;
+    private boolean mediaPreviewEnabled;
+
+    ConversationViewHolder(View itemView,
```

Comment:
this thing is a mess, but it saved a lot of code

Reply:
Should the common parts maybe be abstracted to an even more base view holder?